### PR TITLE
Marketplace: Redirect Partner themes to Congrats page when Activated

### LINKED ISSUE #83876: Partner Themes: Activation not redirecting to the Congrats page
When activating an already installed Partner Theme, its not redirecting to the Congrats page.

Reported: https://github.com/Automattic/wp-calypso/pull/83777#pullrequestreview-1713127679

### PR DESCRIPTION
**Pull request by epeicher**
Related to #83876 

## Proposed Changes

* Remove the `isExternallyManaged` condition so the Partner themes are also sent to the congrats page. After these changes, all the themes will be redirected to the Congrats page when being activated.

### Atomic site
- Navigate to `/themes/:siteId` replacing `:siteId` with an atomic site. 
- Select a Partner theme and click on `Subscribe to activate`
- Complete the checkout process
- In the congrats page, click on `Activate this design`
- Make sure the theme is activated successfully.
- Click on `Back to dashboard`
- Ensure the theme is `Active` as expected

### Non-atomic site
- Repeat the steps in the previous testing adding the plan to the Cart if required
- Make sure the theme is activated as expected

### Any site
- Navigate to the theme list page `/themes/:siteId`
- Find a Partner theme already purchased, e.g. `deco`
- Make sure it is not active, if it is active, activate a different theme
- Click on the three dots menu of the thumbnail on the bottom right
- Click on `Activate`
- Accept the modal
- Ensure the page is redirected to the Congrats page and the theme is activated

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/state/themes/actions/activate.js [hunk 1]
```diff
@@ -96,15 +89,14 @@ export function activate(
...
-		 * and redirect it to the Marketplace theme installation page
-		 */
 		const isDotOrgTheme = !! getTheme( getState(), 'wporg', themeId );
-		if ( isDotOrgTheme && ! isDotComTheme ) {
+		if ( isDotOrgTheme ) {
 			dispatch( productToBeInstalled( themeId, siteSlug ) );
 			return page( `/marketplace/theme/${ themeId }/install/${ siteSlug }` );
 		}
 
+		// Themes should only be either dotCom or dotOrg so this line should never be reached.
```
Comment (cpapazoglou) [inline client/state/themes/actions/activate.js:98]: What about the themes that are both dotCom and dot Org, like `twentytwenty`?

Are they considered both?

Comment (epeicher) [reply client/state/themes/actions/activate.js:98]: They will be considered `dotCom` and they should enter by the first condition

**Comment by epeicher** [comment]
I have reproduced the E2E test failure in my local. I will further investigate it

**Comment by epeicher** [comment]
E2E tests have been fixed after a merge, the item is ready for review now

**Review by WBerredo — APPROVED** [review]
LGTM!

Tested with Partner, Premium and Free themes, all of them redirected to the Congrats page.

**Pull request merged**